common.c

```diff
@@ -58,26 +58,33 @@ void iso_data_free(iso_data_t *iso_data)
     free(iso_data);
 }
 
-choices_t *choices_create(int len)
+choices_t *choices_create(int num_to_allocate)
 {
     choices_t *ret = (choices_t *)calloc(sizeof(choices_t), 1);
     if(!ret) return NULL;
 
-    ret->values = (iso_data_t **)calloc(sizeof(iso_data_t *), len);
+    ret->values = (iso_data_t **)calloc(sizeof(iso_data_t *), num_to_allocate);
     if(!ret->values) {
         free(ret);
         return NULL;
     }
-    ret->len = len;
+    ret->num_allocated = num_to_allocate;
     return ret;
 }
 
-void choices_free(choices_t *c)
+void choices_free(choices_t *choices)
 {
-    if(!c) return;
+    if(!choices) return;
 
-    for(int i = 0; i < c->len; i++) {
-        iso_data_free(c->values[i]);
+    for(int i = 0; i < choices->len; i++) {
+        iso_data_free(choices->values[i]);
+    }
+    free(choices);
+}
+
+void choices_append(choices_t *choices, iso_data_t *data)
+{
+    if(choices->len < choices->num_allocated) {
+        choices->values[choices->len++] = data;
     }
```

Comment:
IMO, we should return an error if we don't have the capacity to add the new item. It feels like a trap to do nothing silently. If the intent is to limit the number of choices, you can still make the caller ignore the error.

Reply:
I didn't ponder this too hard because I planned to do dynamic arrays and not have a hard upper limit.  I'll do some simple checks for now, just in case I don't get around to the dynamic arrays bit.